File: library/src/main/java/com/pengrad/telegrambot/model/request/InputMediaVideo.java
package com.pengrad.telegrambot.model.request;

import com.pengrad.telegrambot.request.ContentTypes;

import java.io.File;
import java.io.Serializable;

/**
 * Stas Parshin
 * 23 November 2017
 */
public class InputMediaVideo extends InputMedia<InputMediaVideo> implements Serializable {
    private final static long serialVersionUID = 1L;

    private Integer width, height, duration;
    private Boolean supports_streaming;
    private Boolean has_spoiler;

    public InputMediaVideo(String media) {
        super("video", media);
    }

    public InputMediaVideo(File media) {
        super("video", media);
    }

    public InputMediaVideo(byte[] media) {
        super("video", media);
    }

    public InputMediaVideo width(Integer width) {
        this.width = width;
        return this;
    }

    public InputMediaVideo height(Integer height) {
        this.height = height;
        return this;
    }

    public InputMediaVideo duration(Integer duration) {
        this.duration = duration;
        return this;
    }

    public InputMediaVideo supportsStreaming(boolean supportsStreaming) {
        this.supports_streaming = supportsStreaming;
        return this;
    }

    public InputMediaVideo hasSpoiler(boolean has_spoiler) {
        this.has_spoiler = has_spoiler;
        return this;
    }

    @Override
    public String getDefaultFileName() {
        return ContentTypes.VIDEO_FILE_NAME;
    }

    @Override
    public String getDefaultContentType() {
        return ContentTypes.VIDEO_MIME_TYPE;
    }
}

File: core/src/main/kotlin/com/pengrad/telegrambot/model/request/InputMediaVideo.kt
package com.pengrad.telegrambot.model.request

import com.pengrad.telegrambot.request.ContentTypes

import java.io.File
import java.io.Serializable

/**
 * Stas Parshin
 * 23 November 2017
 */
class InputMediaVideo : InputMedia<InputMediaVideo>, Serializable {

    private var width: Int? = null
    private var height: Int? = null
    private var duration: Int? = null
    private var supports_streaming: Boolean? = null

    override val defaultFileName: String
        get() = ContentTypes.VIDEO_FILE_NAME

    override val contentType: String
        get() = ContentTypes.VIDEO_MIME_TYPE

    constructor(media: String) : super("video", media) {}

    constructor(media: File) : super("video", media) {}

    constructor(media: ByteArray) : super("video", media) {}

    fun width(width: Int?): InputMediaVideo {
        this.width = width
        return this
    }

    fun height(height: Int?): InputMediaVideo {
        this.height = height
        return this
    }

    fun duration(duration: Int?): InputMediaVideo {
        this.duration = duration
        return this
    }

    fun supportsStreaming(supportsStreaming: Boolean): InputMediaVideo {
        this.supports_streaming = supportsStreaming
        return this
    }

    companion object {
        private const val serialVersionUID = 1L
    }
}
